doc/mgmt/gnmi/save_on_set.md

```diff
@@ -0,0 +1,225 @@
+# Save-On-Set HLD #
+
+_Rev v0.1_
+
+## Table of Contents
+
+- [Save-On-Set HLD](#save-on-set-hld)
+  - [Table of Contents](#table-of-contents)
+    - [Revision](#revision)
+    - [Scope](#scope)
+    - [Definitions/Abbreviations](#definitionsabbreviations)
+    - [Overview](#overview)
+    - [Requirements](#requirements)
+    - [Architecture Design](#architecture-design)
+    - [High-Level Design](#high-level-design)
+      - [Required Changes](#required-changes)
+        - [Telemetry Executable](#telemetry-executable)
+        - [Call to Save ConfigDB](#call-to-save-configdb)
+        - [gNMI.Set() Handler](#gnmiset-handler)
+    - [SAI API](#sai-api)
+    - [Configuration and management](#configuration-and-management)
+      - [gNMI](#gnmi)
+      - [Manifest (if the feature is an Application Extension)](#manifest-if-the-feature-is-an-application-extension)
+      - [CLI/YANG model Enhancements](#cliyang-model-enhancements)
+      - [Config DB Enhancements](#config-db-enhancements)
+    - [Warmboot and Fastboot Design Impact](#warmboot-and-fastboot-design-impact)
+    - [Restrictions/Limitations](#restrictionslimitations)
+    - [Testing Requirements/Design](#testing-requirementsdesign)
+      - [Unit Test cases](#unit-test-cases)
+      - [System Test cases](#system-test-cases)
+    - [Open/Action items - if any](#openaction-items---if-any)
+
+### Revision
+
+| Rev  | Rev Date   | Author(s)          | Change Description |
+|------|------------|--------------------|--------------------|
+| v0.1 | 02/22/2021 | Tomek Madejski (Google), Ryan Lucus (Google)  | Initial version |
+
+### Scope
+
+Add the option for enabling the SONiC gNMI implementation to save its in-memory stored configuration to a file every time it changes.
+
+### Definitions/Abbreviations
+
+- [gNMI](https://github.com/openconfig/reference/blob/master/rpc/gnmi/gnmi-specification.md) - Google Network Management Interface
+- UMF - Unified Management Framework
+
+### Overview
+
+Having configuration be persistant across switch reboot is a useful feature that is not currently implemented by UMF.
+
+The required behaviour is to save the configuration to a file that is used to populate the configuration database during the startup process every time the gNMI.Set() RPC call is performed.
```

Comment:
Why do we need to save the config for every gNMI set operation? Is it possible to periodically save (may be 1 min window) if there is a change in the config from gNMI server?

Reply:
Set is a low frequency operation and delaying the save-on-set would reduce accuracy of saved config if a reset were to happen.